### PR TITLE
Update: set form value and update mutation

### PR DESCRIPTION
**Pull request by puranban**
- Addresses
#2701 
#2702 

* Add query to get data
* Set the value and options in the form 
* Add mutation to update data

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/EditAry/AssessmentRegistryForm/MetadataForm/index.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ import RegionMultiSelectInput, {
 import { PartialFormType } from '../formSchema';
 import StakeholderForm from './StakeholderForm';
 import styles from './styles.css';
+import { BasicOrganization } from '#types';
```
Comment (samshara) [inline app/views/EditAry/AssessmentRegistryForm/MetadataForm/index.tsx]: move absolute imports above relative imports

File: app/views/EditAry/AssessmentRegistryForm/index.tsx [hunk 2]
```diff
@@ -10,6 +10,8 @@ import { EntriesAsList, Error, SetBaseValueArg } from '@togglecorp/toggle-form';
 import MetadataForm from './MetadataForm';
 import { PartialFormType } from './formSchema';
 import styles from './styles.css';
+import { BasicRegion } from '#components/selections/RegionMultiSelectInput';
+import { BasicOrganization } from '#types';
```
Comment (samshara) [inline app/views/EditAry/AssessmentRegistryForm/index.tsx]: move these above relative imports

File: app/views/EditAry/index.tsx [hunk 3]
```diff
@@ -167,20 +173,53 @@ const ENTRIES_FROM_ASSESSMENT = gql`
...
+const ASSESSMENT_REGISTRY_DETAIL = gql`
+    ${ASSESSMENT_REGISTRY_FRAGMENT}
+    query AssessmentDetail (
+        $projectId: ID!,
+        $assessmentId: ID!,
+    ) {
+        project(id: $projectId) {
+            assessmentRegistry(id: $assessmentId) {
+            ...AssessmentRegistryResponse,
+            }
```
Comment (samshara) [inline app/views/EditAry/index.tsx:187]: maybe we need to fetch project id as-well for the caching to work

File: app/views/EditAry/index.tsx [hunk 4]
```diff
@@ -302,6 +345,57 @@ function EditAry(props: Props) {
         },
     );
 
+    const variablesForAssessmentDetails = useMemo(
+        (): AssessmentDetailQueryVariables | undefined => (
+            (assessmentId && projectId) ? { projectId, assessmentId } : undefined
```
Comment (samshara) [inline app/views/EditAry/index.tsx]: Let's use the `isDefined` function to check if the value exists.

File: app/gqlFragments.ts [hunk 5]
```diff
@@ -283,3 +283,75 @@ export const LAST_ACTIVE_PROJECT_FRAGMENT = gql`
...
+                title
+            }
+        }
+        governments {
+            id
+            title
+            verified
+            mergedAs {
+                title
+            }
```
Comment (AdityaKhatri) [inline app/gqlFragments.ts]: use this from fragments

File: app/views/Assessments/index.tsx [hunk 6]
```diff
@@ -142,7 +137,7 @@ function Assessments(props: Props) {
             createStringColumn<Assessment, string>(
                 'title',
                 'Title',
-                (item) => item?.lead?.title ?? item?.leadGroup?.title,
+                (item) => item?.lead?.title,
```
Comment (AdityaKhatri) [inline app/views/Assessments/index.tsx]: lead should be required here

File: app/views/EditAry/AssessmentRegistryForm/formSchema.ts [hunk 7]
```diff
@@ -43,6 +43,7 @@ export const schema: FormSchema = {
         donors: [defaultEmptyArrayType],
         nationalPartners: [defaultEmptyArrayType],
         governments: [defaultEmptyArrayType],
+        lead: [],
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/formSchema.ts]: this should be required

File: app/views/EditAry/index.tsx [hunk 8]
```diff
@@ -302,6 +345,57 @@ function EditAry(props: Props) {
...
+                    });
+
+                    setRegionOptions(result.bgCountries);
+                    setStakeholderOptions([
+                        ...result.leadOrganizations,
+                        ...result.internationalPartners,
+                        ...result.nationalPartners,
+                        ...result.donors,
+                        ...result.governments,
+                    ]);
```
Comment (AdityaKhatri) [inline app/views/EditAry/index.tsx]: need to get unique

File: app/views/EditAry/index.tsx [hunk 9]
```diff
@@ -302,6 +345,57 @@ function EditAry(props: Props) {
...
+            onCompleted: (response) => {
+                if (!response || !response.project?.assessmentRegistry) {
+                    return;
+                }
+
+                const result = removeNull(response?.project?.assessmentRegistry);
+                if (isDefined(result)) {
+                    setValue({
+                        ...result,
+                        lead: result.lead?.id,
```
Comment (AdityaKhatri) [inline app/views/EditAry/index.tsx]: remove conditional ?

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Pull request merged**